jupyterhub/auth.py

```diff
@@ -631,11 +666,51 @@ def _default_executor(self):
         """
      ).tag(config=True)
 
+    admin_groups = Set(
+        help="""
+        User groups that determine admin access. Users not in these groups can still
+        be granted admin status through admin_users. While/blacklisting rules still apply.
+        """
+    ).tag(config=True)
+
     def __init__(self, **kwargs):
         if pamela is None:
             raise _pamela_error from None
         super().__init__(**kwargs)
 
+    @run_on_executor
+    async def is_admin(self, handler, data):
```

Comment:
`@run_on_executor` wraps a blocking method (no `async` here).

Reply:
Thank you. I've fixed that, but now it's a syntax error on the `admin_status = await maybe_future(super().is_admin(handler, data))` in `PAMAuthenticator.is_admin`.

I've tried every combination of `await`, `yield`, and `maybe_future` I can think of and it either is a syntax error, or it's returning a generator instead of the actual result.